android/src/main/java/com/swmansion/gesturehandler/core/FlingGestureHandler.kt

```diff
@@ -35,26 +81,45 @@ class FlingGestureHandler : GestureHandler<FlingGestureHandler>() {
     handler!!.postDelayed(failDelayed, maxDurationMs)
   }
 
-  private fun tryEndFling(event: MotionEvent) = if (
-    maxNumberOfPointersSimultaneously == numberOfPointersRequired &&
-    (
-      direction and DIRECTION_RIGHT != 0 &&
-        event.rawX - startX > minAcceptableDelta ||
-        direction and DIRECTION_LEFT != 0 &&
-        startX - event.rawX > minAcceptableDelta ||
-        direction and DIRECTION_UP != 0 &&
-        startY - event.rawY > minAcceptableDelta ||
-        direction and DIRECTION_DOWN != 0 &&
-        event.rawY - startY > minAcceptableDelta
-      )
-  ) {
-    handler!!.removeCallbacksAndMessages(null)
-    activate()
-    true
-  } else {
-    false
-  }
+  private fun tryEndFling(event: MotionEvent): Boolean {
+
+    fun compareAlignment(
+      vector: Vector,
+      direction: Int,
+      directionVec: Vector = Vector().fromDirection(direction),
+    ): Boolean =
+      vector.computeSimilarity(directionVec) > minDirectionalAlignment &&
+        (this.direction and direction != 0)
```

Comment:
```suggestion
    fun compareAlignment(
      vector: Vector,
      direction: Int,
    ): Boolean =
      (this.direction and direction != 0) &&
        vector.computeSimilarity(Vector().fromDirection(direction)) > minDirectionalAlignment
```

The third argument seems unnecessary here since it's never used.

Bitwise operations will be faster to check than operations on vector, so changing order will save us some processor cycles that we will never notice anyway.

Also, can this also be moved to the `Vector` class? The first argument becomes `this` this way.

Reply:
Current implementation kept `directionVec` but it is now just a reference to a static instance of `Vector`, 
let me know if that is fine, or whether it should still be removed.